src/modules/core/components/Fields/InlineEdit/MultiLineEdit/MultiLineEdit.tsx

```diff
@@ -92,112 +90,86 @@ interface Props {
 const HANDLED = 'handled';
 const NOT_HANDLED = 'not-handled';
 
-class MultiLineEdit extends Component<Props> {
-  static displayName = 'MultiLineEdit';
-
-  static defaultProps = {
-    allowReturns: true,
-    elementOnly: true,
-    readOnly: false,
-    spellCheck: false,
-  };
-
-  editorHasContent = () => {
-    const { $value: editorState } = this.props;
-    const currentContent = editorState.getCurrentContent();
-    return (
-      currentContent.hasText() && currentContent.getPlainText().trim() !== ''
-    );
-  };
-
-  handleReturn = () => {
-    const { allowReturns } = this.props;
-    if (!allowReturns) {
-      return HANDLED;
-    }
-    return NOT_HANDLED;
-  };
-
-  onBlur = (evt: FocusEvent<EditorType>) => {
-    const { onEditorBlur, setValue, $value } = this.props;
-    let editorState = $value;
-    if (!this.editorHasContent()) {
-      editorState = EditorState.push(
-        editorState,
-        ContentState.createFromText(''),
-        'remove-range',
-      );
-      setValue(editorState);
-    }
-    if (onEditorBlur) {
-      onEditorBlur(evt, editorState);
-    }
-  };
-
-  onChange = (editorState: EditorStateType) => {
-    const { setValue } = this.props;
-    setValue(editorState);
-  };
-
-  render() {
-    const {
-      appearance,
-      elementOnly,
-      $error,
-      extra,
-      formatIntl,
-      help,
-      helpValues,
-      $id,
-      label,
-      labelValues,
-      name,
-      placeholder,
-      placeholderValues,
-      readOnly,
-      spellCheck,
-      status,
-      statusValues,
-      $value,
-    } = this.props;
-    const placeholderText =
-      typeof placeholder === 'object'
-        ? formatIntl(placeholder, placeholderValues)
-        : placeholder;
-    return (
-      <div className={getMainClasses(appearance, styles)}>
-        {!elementOnly && (
-          <InputLabel
-            error={$error}
-            help={help}
-            helpValues={helpValues}
-            extra={extra}
-            label={label}
-            labelValues={labelValues}
-          />
-        )}
-        <Editor
-          name={name}
-          editorKey={$id}
-          editorState={$value}
-          handleReturn={this.handleReturn}
-          onBlur={this.onBlur}
-          onChange={this.onChange}
-          placeholder={placeholderText}
-          readOnly={readOnly}
-          spellCheck={spellCheck}
-          stripPastedStyles
-        />
-        <InputStatus
-          status={status}
-          statusValues={statusValues}
+const MultiLineEdit = ({
+  $error,
+  $id,
+  $value,
+  allowReturns = true,
+  appearance,
+  extra,
+  formatIntl,
+  help,
+  helpValues,
+  label,
+  labelValues,
+  name,
+  onEditorBlur,
+  placeholder,
+  placeholderValues,
+  setValue,
+  status,
+  statusValues,
+  elementOnly = true,
+  readOnly = false,
+  spellCheck = false,
+}: Props) => {
+  const prevValue = usePrevious($value);
+  const initialEditorState =
+    $value && prevValue !== $value
```

Comment:
Even though it seems to work, I just thought about this and I think the condition here is wrong. Why should it be empty when the previous value is the same as the current one?